New Resource: `azurerm_durable_task_scheduler`, `azurerm_durable_task_hub`, `azurerm_durable_task_retention_policy` & New Data Source: `azurerm_durable_task_scheduler` (#32194)
terraform-providers/terraform-provider-azurerm: website/docs/r/durable_task_scheduler.html.markdown

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_durable_task_scheduler" "example" {
  name                = "example-durable-task-scheduler"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  sku_name            = "Consumption"
  ip_allow_list       = ["0.0.0.0/0"]

  tags = {
    environment = "production"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Durable Task Scheduler. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Durable Task Scheduler should exist. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the Durable Task Scheduler should exist. Changing this forces a new resource to be created.

* `sku_name` - (Required) The SKU of the Durable Task Scheduler. Possible values include `Consumption` and `Dedicated`. Changing this forces a new resource to be created.

> * `ip_allow_list` - (Required) A list of IP addresses or CIDR ranges that are allowed to access the Durable Task Scheduler.

Reviewer: Required argument ordering, required arguments after `location` must be alphabetically sorted.
```suggestion
* `ip_allow_list` - (Required) A list of IP addresses or CIDR ranges that are allowed to access the Durable Task Scheduler.

* `sku_name` - (Required) The SKU of the Durable Task Scheduler. Possible values include `Consumption` and `Dedicated`. Changing this forces a new resource to be created.
```
Reference: [reference-documentation-standards](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md)

Author: Updated to the canonical phrasing: "Gets information about an existing Durable Task Scheduler."

---

* `capacity` - (Optional) The capacity of the Durable Task Scheduler.

> ~> **Note:** `capacity` is only applicable when `sku_name` is set to `Dedicated`.

Reviewer: Informational note uses the warning note marker `~>`; informational notes should use the informational marker `->`.

```suggestion
-> **Note:** `capacity` is only applicable when `sku_name` is set to `Dedicated`.
```
Reference: [reference-documentation-standards](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md)

Author: Fixed

* `tags` - (Optional) A mapping of tags which should be assigned to the Durable Task Scheduler.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The ID of the Durable Task Scheduler.

* `endpoint` - The endpoint URL of the Durable Task Scheduler.

* `redundancy_state` - The redundancy state of the Durable Task Scheduler.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://developer.hashicorp.com/terraform/language/resources/configure#define-operation-timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Durable Task Scheduler.
* `read` - (Defaults to 5 minutes) Used when retrieving the Durable Task Scheduler.
* `update` - (Defaults to 30 minutes) Used when updating the Durable Task Scheduler.
* `delete` - (Defaults to 30 minutes) Used when deleting the Durable Task Scheduler.

## Import

> Durable Task Schedulers can be imported using the `resource id`, e.g.

Reviewer: The import section must use a singular lead-in statement instead of a plural form:
```suggestion
A Durable Task Scheduler can be imported using the `resource id`, e.g.
```
Reference: [reference-documentation-standards](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md)

Author: Fixed
